libnavigation-util/src/main/java/com/mapbox/navigation/utils/timer/MapboxTimer.kt

```diff
@@ -0,0 +1,31 @@
+package com.mapbox.navigation.utils.timer
+
+import com.mapbox.navigation.utils.thread.ThreadController
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+
+/**
+ * Schedules a delay of [restartAfter] seconds and then restarts.
+ *
+ * @param restartAfter Time delay until the timer should restart
+ * @param listener Hook to receive the events from [MapboxTimerListener]
+ */
+class MapboxTimer(private val restartAfter: Long, private val listener: MapboxTimerListener) {
```

Comment:
This class is targeted to only be used from Kotlin, so how about we accept a lambda instead?

Reply:
Good suggestion 👍